wsi/drm/drmbuffer.cpp

```diff
@@ -89,13 +89,14 @@ void DrmBuffer::Initialize(const HwcBuffer& bo) {
     frame_buffer_format_ = format_;
   }
 
-  FrameBufferManager::GetInstance()->RegisterGemHandles(
-      image_.handle_->meta_data_.num_planes_,
-      image_.handle_->meta_data_.gem_handles_);
+  fb_manager_->RegisterGemHandles(image_.handle_->meta_data_.num_planes_,
+                                  image_.handle_->meta_data_.gem_handles_);
 }
 
-void DrmBuffer::InitializeFromNativeHandle(HWCNativeHandle handle,
-                                           ResourceManager* resource_manager) {
+void DrmBuffer::InitializeFromNativeHandle(
+    HWCNativeHandle handle, ResourceManager* resource_manager,
+    FrameBufferManager* frame_buffer_manager) {
+  fb_manager_ = frame_buffer_manager;
```

Comment:
hmm, this seems redundant. NativeSurface already stores fb_manager_ , it could just pass that along in createframebuffer and createframebufferwithmodifier APIS (And other places we call these API)

Reply:
Hmm, @kalyankondapally  what about line 111? DrmBuffer::Initialize would still need fb_manager_ which runs through this function (InitializeFromNativeHandle) right?

Reply:
you can pass the pointer to that function ?

Reply:
Just seems like a lot of propagation of the frame_buffer_manager parameter everywhere